paper/paper.md

```diff
@@ -0,0 +1,126 @@
+
+__nyaml__ is a Python-based tool with both a command-line interface and an importable API that facilitates the conversion between NXDL XML and a simplified YAML [@YAML2009] representation. YAML's indentation-based syntax enhances human readability and simplifies manual editing. By providing a reliable, lossless round-trip conversion between XML and YAML, __nyaml__ enables developers to edit NeXus definitions more efficiently without sacrificing structural or semantic fidelity.
+
+# Statement of need
+The growth of both the standard and the number of NeXus definition developers makes it all the more important to ensure that the development process is both user-friendly and resilient to errors. The existing representation of NeXus definitions in XML offers structural rigor through the NeXus Definition Language (NXDL) and a well-defined hierarchy for metadata and data types. However, it is verbose and can be difficult to edit by hand. Writing and maintaining NXDL files often involves dealing with deeply nested elements and strict syntax, which can be error-prone and time-consuming, especially for users who are not familiar with XML development. __nyaml__ addresses these challenges by enabling the development of NeXus definitions in a cleaner, YAML-based format while preserving the full structure, semantics, and developer comments of the original XML.
+
+
+# nyaml Converter
+
+The __nyaml__ tool is a Python package developed for converting NeXus application definitions or base classes from YAML format (files with a __.yaml__ extension) into the XML format (files with a __.nxdl.xml__ extension), and vice versa. The package is published on PyPI and therefore can be installed using Python package managers (e.g., __pip__). To write a NeXus application definition or base class in YAML format, the __nyaml__ package introduces a set of keywords and syntactic rules (see [documentation](https://fairmat-nfdi.github.io/nyaml)) that are specific to NXDL (NeXus Definition Language) in YAML.
+
+
+The tool is designed to be used as a command-line utility but can also be utilized as a Python module for programmatic usages. The converter command is invoked using __nyaml2nxdl__, which determines the conversion direction based on the input file (either from YAML to XML or XML to YAML) and delegates the task to the appropriate converter. The converter then executes the corresponding data workflow pipeline (see \autoref{fig:nyaml_workflow}). These workflows are designed to ensure that the conversion process is efficient, reliable, reproducible, and preserves the integrity of the original NeXus data structure and semantics.
```

Comment:
Could start with the sentence introducing the figure, the sentence "These workflows..." is imprecise what makes it efficient? reliable? reproducible? and integrity preserving so should be removed

Reply:
`efficient`: SHA256 hash mechanism makes the conversion efficient, where we build neither the `yaml` object nor the `xml` tree, but produces the `xml` file.

`reliable and reproducible`: The tool can create the same `xml` file if nothing is changed in `yaml` and does not lose any comments.

`integrity`: One can easily extend the existing nexus schemas with the help of the tools that integrates the new physical concept with the existing concepts.

Reply:
Added the name of the workflows to clarify "These workflows..."